src/actions/assetsActions.js

```diff
@@ -517,21 +517,28 @@ function notifyAboutIncreasedBalance(newBalances: Balance[], oldBalances: Balanc
   }
 }
 
-export const fetchAssetsBalancesAction = (showToastIfIncreased?: boolean) => {
-  return async (dispatch: Dispatch, getState: GetState, api: SDKWrapper) => {
-    const {
-      accounts: { data: accounts },
-      balances: { data: balances },
-      featureFlags: { data: { SMART_WALLET_ENABLED: smartWalletFeatureEnabled } },
-    } = getState();
+export const updateAccountBalancesAction = (accountId: string, balances: Balances) => {
+  return async (dispatch: Dispatch, getState: GetState) => {
+    const allBalances = getState().balances.data;
+    const updatedBalances = {
+      ...allBalances,
+      [accountId]: balances,
+    };
+    await dispatch(saveDbAction('balances', { balances: updatedBalances }, true));
```

Comment:
Why wait here for DB to write if we're updating reducer state as well?
```suggestion
    dispatch(saveDbAction('balances', { balances: updatedBalances }, true));
```